internal/binary.bzl

```diff
@@ -84,19 +67,29 @@ def postcss_binary(
         **kwargs: Standard BUILD arguments to pass.
     """
 
-    runner_name = "%s.postcss_runner" % name
+    if stack != None and (len(plugins.keys()) > 0 or len(deps) > 0):
+        fail(ERROR_STACK_PLUGIN_CONFLICT)
 
-    postcss_runner_bin(
-        name = runner_name,
-        deps = deps + plugins.keys(),
-        **dicts.add(kwargs, {"visibility": ["//visibility:private"]})
-    )
+    # If a stack was provided, it acts as the runner. (Plugins come from a
+    # provider created by postcss_stack.)
+    runner_name = stack
+
+    # If a stack wasn't provided, we will be passing through `plugins`.
+    # Manually create the associated `runner`.
+    if stack == None:
+        runner_name = "%s.postcss_runner" % name
+        postcss_runner_bin(
+            name = runner_name,
+            deps = deps + plugins.keys(),
+            **dicts.add(kwargs, {"visibility": ["//visibility:private"]})
+        )
 
     postcss_run(
         name = name,
         src = src,
         output_name = output_name,
         additional_outputs = additional_outputs,
+        plugins_info = ("%s.postcss_plugins" % stack) if stack else None,
```

Comment:
Is there a way to access this at analysis time through a provider, rather than having to rely on a magic target name?

Reply:
I'd love to know if there is one - my Bazel-fu is weak here, and the problem is that postcss_binary is a macro, and postcss_stack is also a macro for a nodejs_binary.

While it was possible in postcss_plugin to pretend to be a Node.js module using `NpmPackageInfo`, I'm not sure a similar approach could work here given we're creating a nodejs_binary and not a 'nodejs_library' of some sort

Reply:
Right now, `postcss_stack` creates a `postcss_plugins_info` and a `postcss_runner_bin`, but those are totally separate targets. I think you can instead have `postcss_plugins_info` be the primary output of `postcss_stack`, and give it an executable `runner` attr that refers to the `postcss_runner_bin`. Then you can access that attribute from `postcss_run`.

I'll try to work this up myself and send a meta-PR if I have time today.

Reply:
I tried to make this work, but I got blocked on some shortcomings of the Node.js rules. Let's land as-is for now and see if we can figure out a way to clean this up afterwards.